Added print utils and print utils calls to reader-writer tests

e-harvey opened the pull request:
@sungeunchoi @jshimek

Commit:
Initial print utils implementation.

	- Implemented init_info, fini_info, print_data,
	and add_line.
	- Updated Makefile.am to reflect these changes.

Commit:
Fixes the cray-tests build.

      - The reader-writer locks PR broke the build, this
      fixes it.

Signed-off-by: Evan Harvey <eharvey@cray.com>

Commit:
Added print utils calls to reader-writer tests.

Signed-off-by: Evan Harvey <eharvey@cray.com>

Commit:
Updated print utils.

-Added print utils api fn calls to reader-writer locks.

-Removed inline keyword from print utils so the linker
can find the print api fns.
-Fixed minor bugs in the print utils.
-Removed csv_header parameter from the print utils.

Signed-off-by: Evan Harvey <eharvey@cray.com>

sungeunchoi commented:
Before you merge this, you should make sure that a445be9 isn't in there twice (it should be merge in separately with #17).

sungeunchoi commented:
Actually a445be9 looks different than the commit in #17.  Not sure if you're able to build this without it tho, so not sure what's going on.

sungeunchoi commented on Makefile.am:
```diff
@@ -2,25 +2,20 @@ AM_CFLAGS = -I$(srcdir)/include
 libcraytests_la_SOURCES = \
-	common/ct_utils.c
+	common/ct_utils.c \
+	common/print_utils.c
```
Given the generic-ness of this name, I'd prefer the files be renamed to ct_print_utils.c and ct_print_utils.h.

sungeunchoi replied:
Similarly, the names of the functions should be prefixed with `ct` and the name of the info struct should be changes to `ct_print_info`.

sungeunchoi commented on common/print_utils.c:
```diff
@@ -0,0 +1,309 @@
+/*******************************************************************************
+ * HELPER MACROS
+ ******************************************************************************/
+#define __CHK_WRITE_ERR(ret, info)					\
```
I'd prefer a `static inline` function here for reasons I've explained in the past.  Plus it's easier to add to.  No backslashes!

sungeunchoi commented on performance/single-node/lock_tests.c:
```diff
@@ -144,40 +144,43 @@ void rwlock_test(unsigned nreaders, unsigned nreads_per_reader,
-
-	printf("%s\n%s\nruntime(s) = %-*gcpu_time = %-*g\n\n", test_name,
-	       test_params, FIELD_WIDTH, ct_wall_clock_time(), FIELD_WIDTH, ct_cpu_time());
+	/* printf("%s\n", test_params); */
```
I think you can get rid of this.

sungeunchoi commented:
I think the internal structure of this needs to be changed to be a little more flexible, but the API is good (and won't change with the changes I'm proposing).  I'll open new issues for that.

Once you fix the things I pointed out above, feel free to merge.

Commit:
Implemented PR18 feedback.

	    - Changed __CHK_WRITE_ERROR to static inline.
	    - Prefixed fns and typedef types with 'ct_'.
	    - Removed unused printfs from lock statements.
	    - Renamed print_utils.{c,h} -> ct_print_utils.{c,h}.

Signed-off-by: Evan Harvey <eharvey@cray.com>
